src/app/api/dashboard/[type]/route.js

```diff
@@ -0,0 +1,184 @@
+import { NextResponse } from "next/server";
+import { db } from "@/utils/firebase";
+import {
+  doc,
+  updateDoc,
+  collection,
+  getDocs,
+  query,
+  where,
+  deleteField,
+  Timestamp,
+} from "firebase/firestore";
+import { authenticate } from "@/utils/auth";
+import { AUTH } from "@/data/dynamic/admin/Admins";
```

Comment:
this should be in a more central location, also the auth for the other ones should be removed